bolt-modules/boltlib/spec/functions/apply_prep_spec.rb

```diff
@@ -282,13 +264,51 @@
     it 'only uses required plugins' do
       facts = Bolt::ResultSet.new(targets.map { |t| Bolt::Result.new(t, value: fact) })
       executor.expects(:run_task)
-              .with(targets, custom_facts_task, includes('plugins'), {})
+              .with(anything, custom_facts_task, includes('plugins'), {})
               .returns(facts)
 
       Puppet.expects(:debug).at_least(1)
       Puppet.expects(:debug).with("Syncing only required modules: non-existing-module.")
-      is_expected.to run.with_params(hostnames.join(','),
-                                     '_required_modules' => ['non-existing-module']).and_return(nil)
+      is_expected.to run.with_params(hostnames,
+                                     '_required_modules' => ['non-existing-module'])
+    end
+  end
+
+  context 'with _catch_errors specified' do
+    let(:custom_facts_task) { Bolt::Task.new('custom_facts_task') }
+    let(:host)              { 'target' }
+    let(:targets)           { inventory.get_targets([host]) }
+    let(:result)            { Bolt::Result.new(targets.first, value: task_result) }
+    let(:resultset)         { Bolt::ResultSet.new([result]) }
+
+    before(:each) do
+      applicator.stubs(:build_plugin_tarball).returns(:tarball)
+      applicator.stubs(:custom_facts_task).returns(custom_facts_task)
+
+      plugins.expects(:get_hook)
+             .with("puppet_agent", :puppet_library)
+             .returns(task_hook)
+    end
+
+    context 'with failing hook' do
```

Comment:
Can this include a test that results are merged as well?

Reply:
I'm having trouble stubbing things and testing the return result, since rspec-puppet only lets you compare equality for the return value. :/

Reply:
Maybe an integration test that runs against a nonexistent target and a real target, and ensures both results are there.

Reply:
D'oh. Added. :P